lib/segment/src/index/hnsw_index/hnsw.rs

```diff
@@ -330,7 +331,7 @@ impl HNSWIndex {
             let level = old_index
                 .as_ref()
                 .and_then(|old_index| old_index.point_level(vector_id))
-                .unwrap_or_else(|| graph_layers_builder.get_random_layer(&mut rng));
+                .unwrap_or_else(|| graph_layers_builder.get_random_layer(rng));
             graph_layers_builder.set_levels(vector_id, level);
```

Comment:
This is the only place during building that I found which was using a random number generator. 

For hnsw gurus: Was I thourough? Are there any other places to pass the `rng` down to?

Reply:
Another possible source of non-determinism is multi-threading, i.e. calling `insert_point()` in `permit.num_cpus` threads.